packages/ai/src/registry.ts

```diff
@@ -12,15 +12,112 @@ import type {
   ModelCapability,
   ModelHandler,
   ModelInfo,
+  ModelMetrics,
   ModelOverride,
+  ModelPricing,
   ModelSelectionWeights,
   ModelSource,
   Provider,
   Providers,
+  RangeConstraint,
   ScoredModel,
   SelectedModel
 } from './types';
 
+// ============================================================================
+// Generic Constraint Helpers
+// ============================================================================
+
+/**
+ * Extracts a numeric value from a complex object for comparison.
+ * Supports nested paths and averaging of multiple values.
+ */
+function extractNumericValue(
```

Comment:
This is a bad approach, remove this function. I need something more flexible, recursive, and generic.